src/transformers/models/led/tokenization_led.py

```diff
@@ -62,80 +62,35 @@ def _pad(
-                This is especially useful to enable the use of Tensor Core on NVIDIA hardware with compute capability
-                >= 7.5 (Volta).
-            return_attention_mask:
-                (optional) Set to False to avoid returning attention mask (default: set to model specifics)
-        """
+        encoded_inputs = super()._pad(
+            encoded_inputs=encoded_inputs,
+            max_length=max_length,
+            padding_strategy=padding_strategy,
+            pad_to_multiple_of=pad_to_multiple_of,
+            return_attention_mask=return_attention_mask,
+        )
+
         # Load from model defaults
         if return_attention_mask is None:
             return_attention_mask = "attention_mask" in self.model_input_names
 
-        required_input = encoded_inputs[self.model_input_names[0]]
-
-        if padding_strategy == PaddingStrategy.LONGEST:
-            max_length = len(required_input)
-
-        if max_length is not None and pad_to_multiple_of is not None and (max_length % pad_to_multiple_of != 0):
-            max_length = ((max_length // pad_to_multiple_of) + 1) * pad_to_multiple_of
-
-        needs_to_be_padded = padding_strategy != PaddingStrategy.DO_NOT_PAD and len(required_input) != max_length
-
-        # Initialize attention mask if not present.
-        if return_attention_mask and "attention_mask" not in encoded_inputs:
-            encoded_inputs["attention_mask"] = [1] * len(required_input)
-
-        if needs_to_be_padded:
-            difference = max_length - len(required_input)
+        if return_attention_mask and "global_attention_mask" in encoded_inputs:
+            required_input = encoded_inputs[self.model_input_names[0]]
+            # `global_attention_mask` need to have the same length as other (sequential) inputs.
+            needs_to_be_padded = len(encoded_inputs["global_attention_mask"]) != len(required_input)
 
-            if self.padding_side == "right":
-                if return_attention_mask:
+            if needs_to_be_padded:
+                difference = len(required_input) - len(encoded_inputs["global_attention_mask"])
 
-                    encoded_inputs["attention_mask"] = encoded_inputs["attention_mask"] + [0] * difference
-                    if "global_attention_mask" in encoded_inputs:
-                        encoded_inputs["global_attention_mask"] = (
-                            encoded_inputs["global_attention_mask"] + [-1] * difference
-                        )
-                if "token_type_ids" in encoded_inputs:
-                    encoded_inputs["token_type_ids"] = (
-                        encoded_inputs["token_type_ids"] + [self.pad_token_type_id] * difference
+                if self.padding_side == "right":
+                    encoded_inputs["global_attention_mask"] = (
+                        encoded_inputs["global_attention_mask"] + [-1] * difference
```

Comment:
Use `-1` because the `0` in `global_attention_mask` is used to indicate `local attention` rather than `not to attend`:

https://github.com/huggingface/transformers/blob/c1000e703b84b4e0f57e204e711e184379f528be/src/transformers/models/led/modeling_led.py#L1536

`global_attention_mask` will be used as follows

https://github.com/huggingface/transformers/blob/c1000e703b84b4e0f57e204e711e184379f528be/src/transformers/models/led/modeling_led.py#L1643-L1653

therefore, `-1` will gvie `0` here for the (merged) `attention_mask`, which makes sense.